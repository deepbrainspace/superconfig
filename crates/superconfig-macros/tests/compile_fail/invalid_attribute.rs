use serde::Serialize;
use superconfig_macros::generate_json_helper;

#[derive(Serialize)]
struct TestStruct {
    value: i32,
}

impl TestStruct {
    #[generate_json_helper(invalid_attribute)]
    pub fn test_method(self) -> Result<Self, String> {
        Ok(self)
    }
}

fn main() {}