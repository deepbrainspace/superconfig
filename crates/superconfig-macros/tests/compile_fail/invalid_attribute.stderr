error: Expected 'in', 'out', 'incoming', 'outgoing', 'auto', 'bidirectional', or 'handle_mode'
  --> tests/compile_fail/invalid_attribute.rs:10:28
   |
10 |     #[generate_json_helper(invalid_attribute)]
   |                            ^^^^^^^^^^^^^^^^^
